submissions/homework_10/cc1759/hw.sql

```diff
@@ -0,0 +1,298 @@
+
+-- -- most and least recent checkout (4)
+-- SELECT MAX(checkout_date) AS newest_checkout FROM checkouts;
+-- SELECT MIN(checkout_date) AS oldlest_checkout FROM checkouts;
+
+-- -- amount of authors (5)
+-- SELECT COUNT(authors.id) AS author_amt FROM authors;
+
+-- -- books published amt per author (6)
+-- SELECT authors.name, COUNT(books.id) AS books_written_amt
+--     FROM authors
+--     LEFT JOIN book_authors ON authors.id = book_authors.author_id
+--     LEFT JOIN books ON book_authors.book_id = books.id
+--     GROUP BY authors.id; 
+
+-- -- checkouts amt per members (7)
+-- SELECT members.full_name, COUNT(checkouts.member_id) AS checkout_amt
+--     FROM members
+--     LEFT JOIN checkouts ON members.id = checkouts.member_id
+--     GROUP BY members.id;
+
+-- -- average rating for each book (8)
+-- SELECT books.title, AVG(book_reviews.rating) AS average_rating
+--     FROM books
+--     LEFT JOIN book_reviews ON books.id = book_reviews.book_id
+--     GROUP BY books.id;
+
+-- -- review amt per book (9)
+-- SELECT books.title, 
+--     GROUP_CONCAT(book_reviews.review ORDER BY book_reviews.review_date ASC SEPARATOR ' | ') AS reviews
+--     FROM books
+--     LEFT JOIN book_reviews ON books.id = book_reviews.book_id
+--     GROUP BY books.id;
+
+-- -- amt of books published per year (10)
+-- SELECT books.year_published, COUNT(books.id)
+--     FROM books
+--     GROUP BY books.year_published;
+
+-- -- number of emails with certain domains (11)
+-- SELECT SUBSTRING(email, INSTR(email, "@") + 1) AS email_domain,
+--     COUNT(id) AS email_amt
+--     FROM web_profiles
+--     GROUP BY id;
+
+-- -- 3 most reviewed books (12)
+-- SELECT books.title AS most_rated_books, COUNT(book_reviews.book_id) AS review_amt
+--     FROM books
+--     JOIN book_reviews ON books.id = book_reviews.book_id
+--     GROUP BY books.id
+--     ORDER BY review_amt DESC
+--     LIMIT 3;
+
+
+-- -- authors with over 1 book published (13)
+-- SELECT authors.name, COUNT(book_authors.author_id) AS books_written_amt
+--     FROM authors
+--     LEFT JOIN book_authors ON authors.id = book_authors.author_id
+
+-- -- highest rated books (14)
+-- SELECT books.title, AVG(book_reviews.rating) AS avg_rating
+--     FROM books
+--     JOIN book_reviews ON books.id = book_reviews.book_id
+--     GROUP by books.id
+--     HAVING AVG(book_reviews.rating) > 4;
+
+-- -- busiest checkout days (15)
+-- SELECT checkout_date AS busy_days, COUNT(checkouts.book_id) AS amt_of_checkouts
+--     FROM checkouts
+--     GROUP BY checkouts.checkout_date
+--     HAVING amt_of_checkouts > 1;
+
+-- -- members with  or more checkouts (16)
+-- SELECT members.full_name, COUNT(checkouts.member_id) AS amt_checkouts
+--     FROM members
+--     JOIN checkouts ON members.id = checkouts.member_id
+--     GROUP BY members.full_name
+--     HAVING amt_checkouts >= 3;
+
+-- -- top 2 members by total checkouts (17)
+-- SELECT members.full_name, COUNT(checkouts.member_id)
```

Comment:
You are missing the key pieces to answer this question. To get only the top 2 you would need to order by the total checkouts and limit to 2 results.